src/analysis/exploration/searchModel.m

```diff
@@ -0,0 +1,155 @@
+function results = searchModel(model,searchTerm,varargin)
+% Search for the specified term in the given model.
+% The performed search is fuzzy if similarity is lower than 1.
+% 
+% USAGE:
+%    results = searchModel(model,searchTerm,...)
+%
+% INPUTS:
+%    model:         The model to search in
+%    searchTerm:    The term to search for
+%    varargin:      Additional parameters as parameter/value pairs or value struct.
+%                   Available parameters are:
+%                    * printLevel - Whether to print the results or just return a search struct. (Default = true) .
+%                    * similarity - Minimum similarity (as provided in calcSim) for matches.
+%                    
+% OUTPUTS:
+%    results:       The results struct build as follows:
+%                    * .field - a struct array either of the basic fields of the model (e.g. rxns, mets etc).
+%                    * .field.id - The id of the matching element
+%                    * .field.matches - information about the matches of the respective ID.
+%                    * .field.matches.value - the value that matched.
+%                    * .field.matches.source - the field that contained the matching value.
+%
+% .. Author: - Thomas Pfau, June 2018
+
+parser = inputParser();
+parser.addParameter('printLevel',1,@isnumeric);
+parser.addParameter('similarity',0.8,@(x) isnumeric(x) && x <= 1 && x >= 0);
+parser.parse(varargin{:});
+similarity = parser.Results.similarity;
+printLevel = parser.Results.printLevel;
+
+%First collect some potential fields, which are known.
+baseFields = {'rxns','mets','vars','ctrs','genes','comps','prots'};
+baseFieldNames = {'reactions', 'metabolites', 'variables', 'constraints', 'genes', 'compartments', 'proteins'};
+nameFields = regexprep(baseFields,'s$','Names');
+knownFields = union(baseFields,nameFields);
+dbFields = getDefinedFieldProperties('DataBaseFields',true);
+knownFields = union(knownFields,dbFields(:,3));
+%And get the annotations which can also be looked up.
+annotationQualifiers = getBioQualifiers();
+results = struct();
+%Now, loop over all basic fields 
+displayedFields = 0;
+for field = 1:numel(baseFields)    
+    cField = baseFields{field};
+    if ~isfield(model,cField)
```

Comment:
May you write this `if` condition as:
```
if isfield(model, cField)
...
end
```
without using `continue`?

Reply:
This will just create another indentation level where it is not really necessary. And there is already quite a deep indentation in the function.